sources/hyper-proxy-shim/src/lib.rs

```diff
@@ -453,48 +406,26 @@ where
                 };
 
                 Box::pin(async move {
+                    #[allow(clippy::never_loop)]
```

Comment:
clippy says this loop never loops, but when I try to remove it things are unhappy. I'm not sure if we want to refactor or just add this clippy exception.

Reply:
It's because `mtry!` is using `break` for flow control, and to do that it has to be in a loop. I'm not sure why `?` doesn't work in this scenario, but replacing `mtry` would be the way to get rid of this "the right way".